app/uk/gov/hmrc/cgtpropertydisposals/repositories/returns/AmendReturnsRepository.scala

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.cgtpropertydisposals.repositories.returns
+
+import cats.data.EitherT
+import cats.syntax.either._
+import com.google.inject.{ImplementedBy, Inject, Singleton}
+import configs.syntax._
+import play.api.Configuration
+import play.api.libs.json._
+import play.modules.reactivemongo.ReactiveMongoComponent
+import reactivemongo.api.Cursor
+import reactivemongo.bson.BSONObjectID
+import reactivemongo.play.json.ImplicitBSONHandlers.JsObjectDocumentWriter
+import uk.gov.hmrc.cgtpropertydisposals.models.Error
+import uk.gov.hmrc.cgtpropertydisposals.models.ListUtils._
+import uk.gov.hmrc.cgtpropertydisposals.models.ids.CgtReference
+import uk.gov.hmrc.cgtpropertydisposals.models.returns.SubmitReturnRequest
+import uk.gov.hmrc.cgtpropertydisposals.repositories.CacheRepository
+import uk.gov.hmrc.cgtpropertydisposals.util.JsErrorOps._
+import uk.gov.hmrc.mongo.ReactiveRepository
+import uk.gov.hmrc.play.http.logging.Mdc.preservingMdc
+
+import scala.concurrent.duration.FiniteDuration
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.control.NonFatal
+
+@ImplementedBy(classOf[DefaultAmendReturnsRepository])
+trait AmendReturnsRepository {
+
+  def fetch(cgtReference: CgtReference): EitherT[Future, Error, List[SubmitReturnRequest]]
+  def save(submitReturnRequest: SubmitReturnRequest, cgtReference: CgtReference): EitherT[Future, Error, Unit]
+
+}
+
+@Singleton
+class DefaultAmendReturnsRepository @Inject() (component: ReactiveMongoComponent, config: Configuration)(implicit
+  val ec: ExecutionContext
+) extends ReactiveRepository[SubmitReturnRequest, BSONObjectID](
+      collectionName = "amend-returns",
+      mongo = component.mongoConnector.db,
+      domainFormat = SubmitReturnRequest.format
+    )
+    with AmendReturnsRepository
+    with CacheRepository[SubmitReturnRequest] {
+
+  //TODO: review parameters
+  val cacheTtl: FiniteDuration  = config.underlying.get[FiniteDuration]("mongodb.draft-returns.expiry-time").value
+  val maxDraftReturns: Int      = config.underlying.get[Int]("mongodb.draft-returns.max-draft-returns").value
```

Comment:
@AndyHWChung are these parameters applicable? How should the house-keeping be done for the entries in this collection?

Reply:
with the way this is being used there shouldn't be a limit to the number of  returns returned for a cgt reference - to correctly display the "recently amended" content we have to know for certain whether the return was recently amended which means looking at everything in this collection for the cgt reference.

The `cacheTtle` needs to be separate from the one for draft returns. The draft return ttl is 30 days, but this ttl needs to be 24 hours. This ttl controls how long a user will see the "recently amended" version of the content on the homepage for after the return has been amended.